text/0431-sagemaker-l2-endpoint.md

```diff
@@ -0,0 +1,1150 @@
+`AWS::SageMaker::EndpointConfig`, the current recommendation of this RFC would be to support the
+specification of both use-cases through a single L2 `EndointConfig` construct. However, this
+presents a challenge when modeling helper APIs like `metricCPUUtilization` or
+`autoScaleInstanceCount` on a related construct as those methods would not universally apply to
+*all* endpoint types.
+
+Given that (1) RFC reviewers may have idiomatic recommendations to solve such modeling challenges
+and (2) the current proposed constructs are still viable for creating a SageMaker endpoint, the
+first draft of this RFC is being published without further revisions accommodating the above newer
+SageMaker features.
+
+[original-pr]: https://github.com/aws/aws-cdk/pull/6107
+[multi-model]: https://aws.amazon.com/blogs/machine-learning/save-on-inference-costs-by-using-amazon-sagemaker-multi-model-endpoints/
+[model-monitor]: https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-amazon-sagemaker-model-monitor/
+[async-inference]: https://aws.amazon.com/about-aws/whats-new/2021/08/amazon-sagemaker-asynchronous-new-inference-option/
+[deployment-guardrails]: https://aws.amazon.com/about-aws/whats-new/2021/11/new-deployment-guardrails-amazon-sagemaker-inference-endpoints/
+[serverless-inference]: https://aws.amazon.com/about-aws/whats-new/2021/12/amazon-sagemaker-serverless-inference/
+
+### What is the high-level project plan?
+
+As the proposed design has been fully implemented in
+[CDK PR #20113](https://github.com/aws/aws-cdk/pull/20113), the delivery timeline of the
+implementation of this RFC will be contingent upon the scope of changes requested by reviewers. For
+baking, the L2 constructs for this module would be marked as experimental, leaving room for further
+adjustments prior to marking the APIs as stable.
+
+### Are there any open issues that need to be addressed later?
+
+1. Please see the [drawbacks section above](#what-are-the-drawbacks-of-this-solution) for potential
+   follow-on work (assuming it need not be incorporated into this RFC).
+1. As observed with [Lambda][lambda-eni-issue] and [EKS][eks-eni-issue], the Elastic Network
+   Interfaces (ENIs) associated with a SageMaker model's VPC are not always cleaned up in a timely
+   manner after downstream compute resources are deleted. As a result, attempts to delete a
+   SageMaker endpoint along with its networking resources (e.g., subnets, security groups) from a
+   CloudFormation stack may cause the stack operation to fail as the ENIs are still in use. From a
+   CDK integration test perspective, specifying `--no-clean` will allow the generation of a snapshot
+   regardless of whether stack deletion will succeed or fail but may hinder snapshot re-generation
+   by subsequent CDK contributors. For this reason, it may be helpful to exclude VPC specification
+   from the endpoint integration test at this time.
```

Comment:
Rather than excluding testing, we should add retries to the tests that take into account these delays.

Reply:
I agree. This is out of scope for the current RFC, but it may give @corymhall some inspiration 😜